packages/frontend/src/components/Timematcher/Timematcher.tsx

```diff
@@ -30,8 +31,8 @@ const generateTimes = (times: [Date, Date]) => {
   const [startTime, endTime] = times;
   timeMap.set(formatTime(startTime), startTime);
   let time = startTime;
-  const FIFTEEN_MINUTES = 15 * 60000;
-  while (time < endTime) {
+  const actualEndTime = new Date(endTime.getTime() - FIFTEEN_MINUTES);
```

Comment:
Why is 15min being subtracted?

Reply:
aha this is kinda dumb but bc each cell on the table represents the time range `celltime -> celltime + 15mins`, the last cell on the table should be 15 mins before the actual end time (e.g. if end time is five, the last cell should be 4:45)